### ISSUE TITLE
model that achieves best validation loss not used in evaluating performance in train mode

### ISSUE THREAD

**Issue by StevenSong** [broadinstitute/ml4h#364]
**What**
only the model that achieves the best validation loss should be saved and used in evaluating performance. currently, only the model that achieves best val loss is saved. however, the model returned by `train_model_from_generators` and used for predicting/evaluating test data is not that model. the model returned is from the end of training - either when early stopping is triggered or when epoch limit is reached.

in `models.py`: https://github.com/broadinstitute/ml/blob/48a397cb33c5670f23926e5ccea57f63fb8fbf63/ml4cvd/models.py#L1058-L1130

in `recipes.py`:
https://github.com/broadinstitute/ml/blob/48a397cb33c5670f23926e5ccea57f63fb8fbf63/ml4cvd/recipes.py#L138-L148

relevant keras issue: https://github.com/keras-team/keras/issues/2768

`model.fit()` in `train_model_from_generators` does not save and return the best model by default. saving the best model can be configured with callback `ModelCheckpoint` which is done in `_get_callbacks()`. however, with this approach, the model still needs to be loaded after training, as [in this example](https://github.com/keras-team/keras/issues/2768#issuecomment-221919149).

an alternative is to configure `EarlyStopping` with argument `restore_best_weights = True`, however this is perhaps not always correct [as I noted here](https://github.com/keras-team/keras/issues/2768#issuecomment-660157917)

**Why**
the reported performance metrics should reflect the best model

**How**
explicitly load model weights before returning model

**Acceptance Criteria**
model that achieves best validation loss is used for reporting perf metrics on test data
